src/WindowWrapper.cpp

```diff
@@ -0,0 +1,341 @@
+#include <Gosu/Gosu.hpp>
+
+namespace Gosu
+{
+  class WindowForWrapper : public Gosu::Window
+  {
+  public:
+    WindowForWrapper(int width, int height, bool fullscreen, double update_interval, bool resizable);
+    void set_draw(void function(void *data), void *data);
+    void set_update(void function(void *data), void *data);
+    void set_button_down(void function(void *data, unsigned btn), void *data);
+    void set_button_up(void function(void *data, unsigned btn), void *data);
+    void set_drop(void function(void *data, const char *filename), void *data);
+    void set_needs_redraw(bool function(void *data), void *data);
+    void set_needs_cursor(bool function(void *data), void *data);
+    void set_close(void function(void *data), void *data);
+
+    void update() override;
+    void draw() override;
+    void gosu_button_down(unsigned btn); // Enables fullscreen toggle
```

Comment:
Hmmm, maybe clearer to name this handle_fullscreen_shortcuts or default_button_down?